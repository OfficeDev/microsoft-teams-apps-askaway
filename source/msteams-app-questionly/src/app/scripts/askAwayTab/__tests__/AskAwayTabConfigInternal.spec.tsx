/**
 * @jest-environment jsdom
 */

import { Flex, Image, Text } from '@fluentui/react-northstar';
import { configure, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';
import AskAwayTabConfigInternal from '../AskAwayTabConfigInternal';

configure({ adapter: new enzymeAdapterReact16() });
jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
        };
    },
}));

describe('AskAwayTabConfigInternal Component', () => {
    afterAll(() => {
        jest.resetAllMocks();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<AskAwayTabConfigInternal />);

        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<AskAwayTabConfigInternal />);

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(1);
        expect(component.find(Image)).toHaveLength(1);
    });
});
