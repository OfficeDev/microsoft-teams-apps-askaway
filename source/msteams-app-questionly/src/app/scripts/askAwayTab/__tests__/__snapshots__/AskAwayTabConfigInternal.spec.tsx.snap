// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskAwayTabConfigInternal Component should match the snapshot 1`] = `
<Flex
  className="tab-config-container"
  column={true}
  gap="gap.large"
  hAlign="center"
  vAlign="center"
>
  <Image
    accessibility={[Function]}
    alt="image"
    as="img"
    src="askaway_tab_added.png"
  />
  <FlexItem
    align="center"
  >
    <Text
      as="span"
      content="tab.configTabText"
      size="large"
      weight="bold"
    />
  </FlexItem>
</Flex>
`;
